dll/win32/browseui/internettoolbar.cpp

```diff
@@ -1001,19 +1017,50 @@ HRESULT STDMETHODCALLTYPE CInternetToolbar::GetClassID(CLSID *pClassID)
     return S_OK;
 }
 
+HRESULT CInternetToolbar::SetDirty()
+{
+    if (fIgnoreChanges)
+        return S_OK;
+    IUnknown_Exec(fSite, CGID_ShellBrowser, IDM_NOTIFYITBARDIRTY, 0, NULL, NULL);
+    return S_OK;
+}
+
 HRESULT STDMETHODCALLTYPE CInternetToolbar::IsDirty()
 {
     return E_NOTIMPL;
 }
 
 HRESULT STDMETHODCALLTYPE CInternetToolbar::Load(IStream *pStm)
 {
-    return E_NOTIMPL;
+    fIgnoreChanges = TRUE;
+    HRESULT hr = InitNew();
+    ITBARSTATE state;
+    if (SUCCEEDED(hr))
+    {
+        hr = S_FALSE;
+        ULONG cb = sizeof(state);
+        if (pStm->Read(&state, cb, &cb) == S_OK && state.Signature == state.SIG)
```

Comment:
Should there also be a check on `cb == sizeof(state)` ?

Reply:
At some point when it starts saving itself in the same place as Windows, yes. For now, it does not really matter, the signature must be removed when it becomes Windows compatible. The stream is over 500 bytes on XP.